standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -642,112 +642,140 @@ public void rollbackTransaction() {
   @Override
   public void createCatalog(Catalog cat) throws MetaException {
     LOG.debug("Creating catalog {}", cat);
-    boolean committed = false;
+
     MCatalog mCat = catToMCat(cat);
+
+    final Transaction tx = pm.currentTransaction();
+
     try {
-      openTransaction();
+      tx.begin();
       pm.makePersistent(mCat);
-      committed = commitTransaction();
+      tx.commit();
     } finally {
-      if (!committed) {
-        rollbackTransaction();
+      if (tx.isActive()) {
+        tx.rollback();
       }
     }
   }
 
   @Override
   public void alterCatalog(String catName, Catalog cat)
-      throws MetaException, InvalidOperationException {
+      throws MetaException, InvalidOperationException, NoSuchObjectException {
```

Comment:
This should also throw NoSuchObjectException if user is attempting to alter a non-existing catalog